packages/connect-query-core/src/index.ts

```diff
@@ -0,0 +1,25 @@
+// Copyright 2021-2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+export type { ConnectQueryKey } from "./connect-query-key.js";
+export { createConnectQueryKey } from "./connect-query-key.js";
+export { createProtobufSafeUpdater } from "./utils.js";
+export type { ConnectUpdater } from "./utils.js";
+export { callUnaryMethod } from "./call-unary-method.js";
+export { createInfiniteQueryOptions } from "./create-infinite-query-options.js";
+export type { ConnectInfiniteQueryOptions } from "./create-infinite-query-options.js";
+export { createQueryOptions } from "./create-query-options.js";
+export { addStaticKeyToTransport } from "./transport-key.js";
+export type { SkipToken } from "@tanstack/query-core";
+export { skipToken } from "@tanstack/query-core";
```

Comment:
`skipToken` is used directly at runtime to inform behaviour within connect-query. To me, that's enough of a reason to re-export. To be fair, that's still inconsistent, since we still rely on `QueryClientProvider` but it's a good first step to being a little easier to see the full surface API.

Reply:
So we're only going to re-export things that are frequently exported. Seems reasonable 👍